packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -373,6 +375,16 @@ - (NSNumber *)getDesiredImageQuality:(NSNumber *)imageQuality {
   return imageQuality;
 }
 
+- (void)presentationControllerDidDismiss:(UIPresentationController *)presentationController {
+  dispatch_queue_t backgroundQueue =
+      dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0);
+  dispatch_async(backgroundQueue, ^{
+    self.result(nil);
+    self.result = nil;
+    self->_arguments = nil;
+  });
+}
```

Comment:
Results should be called on the main thread.

I would also check if `self.result` is not nil.

Reply:
Fixed